app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -82,10 +84,14 @@ class RecentSyncedTabFeature(
 
     // UI will either not be displayed if not authenticated (DefaultPresenter.start),
     // or the display state will be tied directly to the success and error cases.
-    override fun stopLoading() = Unit
+    override fun stopLoading() {
+        isSyncing = false
+    }
 
     override fun onError(error: SyncedTabsView.ErrorType) {
-        store.dispatch(AppAction.RecentSyncedTabStateChange(RecentSyncedTabState.None))
+        if ((!isSyncing && error == SyncedTabsView.ErrorType.NO_TABS_AVAILABLE) || error.isFatal()) {
```

Comment:
Instead of having an `isSyncing` variable, could we check `store.state.recentSyncedTabState` to see if it's loading? Not saying this is necessarily the correct thing to do, but in prior reviews I had an inkling that maybe `recentSyncedTabState` should be tri-state with `stopLoading` implemented.